tools/cmake/toolchains/ARM.cmake

```diff
@@ -8,49 +8,39 @@ set(CMAKE_AR "armar")
 set(ARM_ELF2BIN "fromelf")
 set_property(GLOBAL PROPERTY ELF2BIN ${ARM_ELF2BIN})
 
-# Sets toolchain options
-function(mbed_set_toolchain_options target)
-    list(APPEND common_options
-        "-c"
-        "--target=arm-arm-none-eabi"
-        "-mthumb"
-        "-Wno-armcc-pragma-push-pop"
-        "-Wno-armcc-pragma-anon-unions"
-        "-Wno-reserved-user-defined-literal"
-        "-Wno-deprecated-register"
-        "-fdata-sections"
-        "-fno-exceptions"
-        "-fshort-enums"
-        "-fshort-wchar"
-    )
+# tell cmake about compiler targets.
+# This will cause it to add the --target flag.
+set(CMAKE_C_COMPILER_TARGET arm-arm-none-eabi)
+set(CMAKE_CXX_COMPILER_TARGET arm-arm-none-eabi)
 
-    target_compile_options(${target}
-        INTERFACE
-            $<$<COMPILE_LANGUAGE:C>:${common_options}>
-    )
-
-    target_compile_options(${target}
-        INTERFACE
-            $<$<COMPILE_LANGUAGE:CXX>:${common_options}>
-    )
+# Sets toolchain options
+list(APPEND common_options
+    "-mthumb"
```

Comment:
`-c` was removed as it's not required, just checking if it was intentionally removed.

Reply:
Yep that was intentional, CMake will add -c as needed.